packages/editor/src/components/post-actions/actions.js

```diff
@@ -694,40 +723,47 @@ const resetTemplateAction = {
 		onActionPerformed,
 	} ) => {
 		const [ isBusy, setIsBusy ] = useState( false );
-		const { revertTemplate } = unlock( useDispatch( editorStore ) );
+		const { revertTemplate, removeTemplates } = unlock(
+			useDispatch( editorStore )
+		);
 		const { saveEditedEntityRecord } = useDispatch( coreStore );
 		const { createSuccessNotice, createErrorNotice } =
 			useDispatch( noticesStore );
 		const onConfirm = async () => {
 			try {
-				for ( const template of items ) {
-					await revertTemplate( template, {
-						allowUndo: false,
-					} );
-					await saveEditedEntityRecord(
-						'postType',
-						template.type,
-						template.id
+				if ( items[ 0 ].type === TEMPLATE_PART_POST_TYPE ) {
+					await removeTemplates( items );
+				} else {
+					for ( const template of items ) {
+						if ( template.type === TEMPLATE_POST_TYPE ) {
```

Comment:
Why do we need all the post type checks. Or in other words, why reverting changes is different between template, template parts. And I don't see where we're handling patterns?

Reply:
> Why do we need all the post type checks. Or in other words, why reverting changes is different between template, template parts. And I don't see where we're handling patterns?

There was is revert for patterns that was already the case before, one can not change the original theme patterns only duplicate and create a new version. Regarding the difference between reverting a template and reverting a template part. When we revert a template part we simply delete the user post storing from the database, when we revert a template we make an API request to the server to retrieve the original blocks in that template and store those original blocks as the user content. I don't know why we do things differently in both cases, we are doing it like that for a long time.

Reply:
> Regarding the difference between reverting a template and reverting a template part. When we revert a template part we simply delete the user post storing from the database, when we revert a template we make an API request to the server to retrieve the original blocks in that template and store those original blocks as the user content. I don't know why we do things differently in both cases, we are doing it like that for a long time.

I think we should remove the post in both cases to be honest.

> There was is revert for patterns that was already the case before, one can not change the original theme patterns only duplicate and create a new version.

Why do we allow the revert action for patterns then (you added `|| isPattern`)

Reply:
> Why do we allow the revert action for patterns then (you added || isPattern)

I thought it was possible to see patterns with template parts but that does not seems to be the case so I removed is `|| isPattern`.

Reply:
@jorgefilipecosta Let's track this in an issue or something (may be good to try to ping the folks that introduced this in the first place for more info/context and why it's not the same)